From 140aeb3949039cb7df43ac2d7e4a755fb4711879 Mon Sep 17 00:00:00 2001
From: Ian Lynagh <igloo@earth.li>
Date: Tue, 19 Oct 2010 13:27:27 +0000
Subject: [PATCH] Fix a retainer profiling segfault The bitmap type wasn't big
 enough to hold large bitmaps on 64 bit platforms. Profiling GHC was
 segfaulting when retainStack was handling a size 33 bitmap.

---
 rts/RetainerProfile.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rts/RetainerProfile.c b/rts/RetainerProfile.c
index e80a588e86f..c5a7bf78977 100644
--- a/rts/RetainerProfile.c
+++ b/rts/RetainerProfile.c
@@ -1270,7 +1270,7 @@ retainStack( StgClosure *c, retainer c_child_r,
     stackElement *oldStackBoundary;
     StgPtr p;
     StgRetInfoTable *info;
-    StgWord32 bitmap;
+    StgWord bitmap;
     nat size;
 
 #ifdef DEBUG_RETAINER
